kedro-airflow/features/airflow.feature

```diff
@@ -8,21 +8,20 @@ Feature: Airflow
     And I have prepared a config file
     And I have run a non-interactive kedro new
     And I have prepared a data catalog
-    And I have executed the kedro command "airflow create -t ../airflow/dags/"
+    And I have executed the kedro command "airflow create -t ../airflow/dags/project_dummy_dag.py"
     When I execute the airflow command "tasks list project-dummy"
     Then I should get a successful exit code
     And I should get a message including "predict"
     And I should get a message including "report"
     And I should get a message including "split"
-    And I should get a message including "train"
 
   Scenario: Run Airflow task locally with latest Kedro
     Given I have installed kedro version "latest"
     And I have prepared a config file
     And I have run a non-interactive kedro new
     And I have prepared a data catalog
-    And I have executed the kedro command "airflow create -t ../airflow/dags/"
+    And I have executed the kedro command "airflow create -t ../airflow/dags/project_dummy_dag.py"
```

Comment:
It isn't immediately obvious to me, why does it has to break, shouldn't the default work as the same?

Reply:
Default is fine but if there is previous code that uses the -t option to provide a directory instead of dir+filepath than this would break now. I could check if its a dir or a file and handle those cases in code 🤔

Reply:
I am slightly inclined to just have --target, whether it is path/dir should just work. But I would love to hear comments from others. And definitely less breaking changes are better.

Reply:
My thoughts on this are:
* being able to change output filename is a relatively small new feature, since it can already be achieved by running `kedro airflow create && mv ...` (unlike the `jinja-file` option, which can't currently be achieved so easily)
* as per @noklam, it's nice to do non-breaking changes where possible, especially when it's just to add a relatively minor feature. So we should not release in 0.5.1 a version that changes the name of the `--target-dir` flag to something else or which breaks current behaviour that this argument can take a directory

In an ideal world, Nok's idea of having a `--target` option that can take a directory or a filepath sounds like the right solution to me. The only breaking part of that is renaming the `--target-dir` flag since it's not just a directory any more. Technically we should do it as follows:
1. introduce in 0.5.1 a warning that the argument is changing name
2. change the argument name _in 0.6.0_

But this is a bit of a faff  for such a minor change to a plugin that not that many people use. (Can we check how many people use `kedro airflow create` with the `--target-dir` flag on telemetry? I think it's going to be very small... @datajoely) So I'd be tempted to just skip step 1 above.

Then `--target` can be a directory or filename. The default behaviour when no target is specified should remain as it currently is. If `--target` supplies a directory, the output file should be in that directory with the default `{package_name}_dag.py` name. If `--target` supplies a file, then save to that file (creating any parent directories that are required).

Having an (optional) user-specified filename also makes more sense to me than the initial proposal where the output file was labelled according to the pipeline name (e.g. why don't we also label it according to `env`?).

All this is becoming a bit more complicated a change so I'd also advise splitting it off into a separate PR and just focussing on the `jinja-file` addition here 🙂

Reply:
I don't mind so much, happy to fork it off into a separate PR. I found being able to either provide a dir or filepath more intuitive than just a dir with a "default" filename and then using mv afterwards. @AntonyMilneQB @noklam would you still be happy to merge that PR relatively quickly or is it a too minor change?

Reply:
"quickly" in terms of the PR not necessarily as a release :)

Reply:
It's a minor change but it does make sense as a new argument, so if you think it's useful then I'd be very happy to accept such a PR. @noklam is my above description of a `--target` flag that takes a file or a directory what you were thinking?

Also I suspect that once the PRs are merged we'll do a release pretty soon after anyway, unless we're waiting for any other particular changes to go in.

Reply:
I am thinking the same about the `--target` flag.